File: 506. Relative Ranks/RelativeRanks.cs
public class Solution {
    public string[] FindRelativeRanks(int[] score) {
        string[] result = new string[score.Length];

        int[] idxs = score.Select((value, index) => new { Value = value, Index = index })
                          .OrderByDescending(v => v.Value)
                          .Select(v => v.Index)
                          .ToArray();
                          
        for (int i = 0; i < idxs.Length; i++) {
            switch (i) {
                case 0:
                    result[idxs[i]] = "Gold Medal";
                    break;
                case 1:
                    result[idxs[i]] = "Silver Medal";
                    break;
                case 2:
                    result[idxs[i]] = "Bronze Medal";
                    break;
                default:
                    result[idxs[i]] = (i + 1).ToString();
                    break;
            }
        }
        
        return result;
    }
}

File: 506. Relative Ranks/relative_ranks.py
class Solution(object):
    def findRelativeRanks(self, score):
        """
        :type score: List[int]
        :rtype: List[str]
        """
        sorted_scores = sorted([(s, i) for i, s in enumerate(score)], reverse=True)

        result = ["" for _ in score]

        for rank, (s, i) in enumerate(sorted_scores, 1):
            if rank == 1:
                result[i] = "Gold Medal"
            elif rank == 2:
                result[i] = "Silver Medal"
            elif rank == 3:
                result[i] = "Bronze Medal"
            else:
                result[i] = str(rank)
        
        return result